tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -755,7 +755,7 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
         PO.SkipIntroducerKeywords = true;
         PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
         XMLEscapingPrinter Printer(RelDeclsStream);
-        if (BaseType) {
+        if (BaseType && RelatedDecl->getDeclContext() == BaseType->getAnyNominal()) {
```

Comment:
If the base type is an inherited protocol or a subclass the 'if' statement check in this patch will still fail though.

When you first find the related decl, can you see if it is being called on the current method's 'self' value, and set a BaseType there; otherwise you know it's from a different context. Then you could still print substituted types for subclasses and protocols.